pkg/validate.go

```diff
@@ -50,3 +50,70 @@ func ValidateNetworkChaos(testsDetails *types.TestDetails, TargetPodIP, HelperPo
 	}
 	return nil
 }
+
+// ValidateResourceChaos will validate the actual cpu injection on the application pod
+// It will compare the cpu usage before and after chaos
+func ValidateResourceChaos(validation bool, resourceName, namespace, expName string) error {
+
+	var cpuUsage int
+	var memoryUsage int
+
+	// waiting for chaos to progress
+	klog.Info("[Wait]: Waiting for chaos to progress...")
+	time.Sleep(60 * time.Second)
+
+	switch expName {
+
+	case "pod-cpu-hog":
+		cpuUsage, err = GetCPUUsage(validation, resourceName, namespace, "pod")
+		if err != nil {
+			return errors.Errorf("fail to get the cpu usage after chaos err: %v", err)
+		}
+		klog.Infof("[CPU]: The CPU usage of app pod after chaos is: %vm", cpuUsage)
+		// for default cpu spike of half core
+		if cpuUsage < 500 {
```

Comment:
is it always accurate 500+? or would you like to add some +-20/50 by assuming some difference? and do we also need a check for upper bound? usage shouldn't be more than some value?